deeplake/core/vectorstore/test_deeplake_vectorstore.py

```diff
@@ -1769,3 +1769,8 @@ def test_uuid_fix(local_path):
     vector_store.add(text=texts, id=ids, embedding=embeddings, metadata=metadatas)
 
     assert vector_store.dataset.id.data()["value"] == list(map(str, ids))
+
+
+def test_read_only():
+    db = VectorStore("hub://davitbun/twitter-algorithm")
+    assert db.dataset.read_only == True
```

Comment:
Where is this flag set to be True?

Reply:
it's because the dataset is public, and testing account doesn't have write access to it

Reply:
Sorry for the confusion, I was just wondering where in the code the flag was set to True.